addons/website_sale/models/website.py

```diff
@@ -350,11 +345,10 @@ def sale_get_order(self, force_create=False, update_pricelist=False):
             if sale_order_sudo.fiscal_position_id != previous_fiscal_position:
                 sale_order_sudo.order_line._compute_tax_id()
 
-            if sale_order_sudo.pricelist_id != previous_pricelist:
-                update_pricelist = True
-        elif update_pricelist:
-            # Only compute pricelist if needed
-            pricelist_id = self._get_current_pricelist_id(partner_sudo)
+        # check for any modification of pricelist (added, removed or changed)
+        previous_pricelist = sale_order_sudo.pricelist_id.id
+        pricelist_id = self._get_current_pricelist_id(partner_sudo)
```

Comment:
If update_pricelist is False and partner wasn't modified, we don't have to fetch the current pricelist :thinking: